src/services/rfqm_service.ts

```diff
@@ -497,7 +500,24 @@ export class RfqmService {
             await this.processRfqmJobAsync(job.orderHash, workerAddress);
         }
 
-        return this._blockchainUtils.isWorkerReadyAsync(workerAddress, gasPrice);
+        const balance = await this._blockchainUtils.getAccountBalanceAsync(workerAddress);
+        const isWorkerReady = await this._blockchainUtils.isWorkerReadyAsync(workerAddress, balance, gasPrice);
+
+        if (!isWorkerReady) {
+            return false;
+        }
+
+        // Publish a heartbeat if the worker is ready to go
+        try {
+            if (RFQM_WORKER_INDEX === undefined) {
+                throw new Error('Worker index is undefined');
+            }
+            await this._dbUtils.upsertRfqmWorkerHeartbeatToDbAsync(workerAddress, RFQM_WORKER_INDEX, balance);
+        } catch (e) {
+            logger.error({ workerAddress, balance }, `Worker failed to write a heartbeat to storage: ${e}`);
```

Comment:
If this happens, do we not want to return `false`?

Reply:
Mmm...I didn't want the worker to not start up and possibly not mine transactions it could have, so I elected to just log it.

Reply:
I see, as long as we index on this error in one way or another, it will be useful